tests/testthat/test-apply_theme.R

```diff
@@ -0,0 +1,407 @@
+#' T2.7 The fontfamily applied through `visR::apply_theme()` is used in the resulting `ggplot` object.
+#' T2.8 The grid applied through `visR::apply_theme()` is used in the resulting `ggplot` object.
+#' T2.9 The background applied through `visR::apply_theme()` is used in the resulting `ggplot` object.
+#' T2.10 The legend_position applied through `visR::apply_theme()` is used in the resulting `ggplot` object.
+#' T2.11 The legend_position defined in `visR::visr()` is correctly passed through to the resulting `ggplot` object.
+
+# Requirement T1 ---------------------------------------------------------------
+
+testthat::context("apply_theme - T1. The `define_theme()` function returns a `visR_theme` object can contain valid input parameters for `apply_theme()`.")
+
+testthat::test_that("T1.1 No error when no parameters are specified.",{
+  
+  testthat::expect_error(visR::define_theme(), NA)
+  
+})
+
+testthat::test_that("T1.2 Not specifying any parameters returns a list.",{
+  
+  theme <- visR::define_theme()
+  
+  testthat::expect_true(is.list(theme))
+  
+})
+
+testthat::test_that("T1.3 No error when `strata` is `NULL`.",{
+  
+  testthat::expect_error(visR::define_theme(strata = NULL), NA)
+  
+})
+
+testthat::test_that("T1.4 A warning when `strata` is an empty `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(strata = list()))
+  
+})
+
+testthat::test_that("T1.5 A warning when `strata` is an unnamed `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(strata = list("v", "i", "s", "R")))
+  
+})
+
+testthat::test_that("T1.6 No warning when `strata` is a named `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(strata = list("visR" = "visR")), NA)
+  
+})
+
+testthat::test_that("T1.7 No error when `fontsizes` is `NULL`.",{
+  
+  testthat::expect_error(visR::define_theme(fontsizes = NULL), NA)
+  
+})
+
+testthat::test_that("T1.8 A warning when `fontsizes` is an empty `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(fontsizes = list()))
+  
+})
+
+testthat::test_that("T1.9 A warning when `fontsizes` is an unnamed `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(fontsizes = list("v", "i", "s", "R")))
+  
+})
+
+testthat::test_that("T1.10 No warning when `fontsizes` is a named `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(fontsizes = list("visR" = "visR")), NA)
+  
+})
+
+testthat::test_that("T1.11 A message when `fontsizes` is a numerical value.",{
+  
+  testthat::expect_message(visR::define_theme(fontsizes = 12))
+  
+})
+
+testthat::test_that("T1.12 A warning when `fontsizes` is neither `NULL`, a `list` or a `numeric`.",{
+  
+  testthat::expect_warning(visR::define_theme(fontsizes = "visR"))
+  
+})
+
+testthat::test_that("T1.13 No error when `fontfamily` is a string.",{
+  
+  testthat::expect_error(visR::define_theme(fontfamily = "Times"), NA)
+  
+})
+
+testthat::test_that("T1.14 A warning when `fontfamily` is anything but a string.",{
```

Comment:
what if fontfamily is a list of strings?

Reply:
Good catch, fixed it